Makefile

```diff
@@ -16,52 +22,70 @@ else
 PACK_VER:=$(VERSION)
 endif
 
-BINDIR=$(PWD)/install
+BINDIR=$(PWD)/install-$(ENGINE)
+
+BACKENDS_PATH ?= $(BINDIR)/backends
+
+#----------------------------------------------------------------------------------------------
+
+setup:
+	@echo Setting up system...
+	$(SHOW)./deps/readies/bin/getpy
+	$(SHOW)./system-setup.py
+
+.PHONY: all setup build clean deps pack pack_ramp pack_deps test
+
+all: build
 
-.PHONY: all clean deps pack rlec_runpath_fix
+ifeq ($(DEBUG),1)
+CMAKE_FLAGS += -DCMAKE_BUILD_TYPE=Debug
+endif
 
-all:
-ifeq ($(wildcard build/.),)
+build:
+ifeq ($(wildcard build/Makefile),)
 	mkdir -p build
 	cd build; \
-	cmake -DDEPS_PATH=../deps/install ..
+	cmake -DENGINE=$(ENGINE) -DDEPS_PATH=../deps/install-$(ENGINE) $(CMAKE_FLAGS) ..
```

Comment:
Let's rename `ENGINE` to `DEVICE`, as the term engine it might be confusing.

This applies to all changes, even prior to this comment.

Reply:
Yes, of course. Actually, don't where this ENGINE came up from. Should have been DEVICE in the  first place.